rosidl_generator_cpp/test/test_traits.cpp

```diff
@@ -13,14 +14,230 @@
 // limitations under the License.
 
 #include <gtest/gtest.h>
+// #include "rosidl_generator_cpp/idl/idl_only_types.hpp"
+#include "rosidl_generator_cpp/msg/defaults.hpp"
 #include "rosidl_generator_cpp/msg/empty.hpp"
+#include "rosidl_generator_cpp/msg/bounded_sequences.hpp"
+#include "rosidl_generator_cpp/msg/nested.hpp"
 #include "rosidl_generator_cpp/msg/strings.hpp"
+#include "rosidl_generator_cpp/msg/w_strings.hpp"
 #include "rosidl_generator_cpp/srv/empty.hpp"
 
 using rosidl_generator_traits::is_message;
 using rosidl_generator_traits::is_service;
 using rosidl_generator_traits::is_service_request;
 using rosidl_generator_traits::is_service_response;
+using rosidl_generator_traits::to_yaml;
+
+TEST(Test_rosidl_generator_traits, to_yaml) {
+  {
+    const rosidl_generator_cpp::msg::Empty msg;
+    EXPECT_STREQ("null\n", to_yaml(msg).c_str());
+  }
+
+  {
+    rosidl_generator_cpp::msg::Defaults msg;
+    msg.float64_value = 1.0;
+    EXPECT_STREQ(
+      R"(bool_value: true
+byte_value: 0x32
+char_value: 100
+float32_value: 1.12500
+float64_value: 1.00000
+int8_value: -50
+uint8_value: 200
+int16_value: -1000
+uint16_value: 2000
+int32_value: -30000
+uint32_value: 60000
+int64_value: -40000000
+uint64_value: 50000000
+)",
+      to_yaml(
+        msg).c_str());
+  }
+
+  {
+    rosidl_generator_cpp::msg::Strings msg;
+    msg.string_value = "Hello\nworld";
+    EXPECT_STREQ(
+      R"(string_value: "Hello
+world"
+string_value_default1: "Hello world!"
+string_value_default2: "Hello'world!"
+string_value_default3: "Hello\"world!"
+string_value_default4: "Hello'world!"
+string_value_default5: "Hello\"world!"
+bounded_string_value: ""
+bounded_string_value_default1: "Hello world!"
+bounded_string_value_default2: "Hello'world!"
+bounded_string_value_default3: "Hello\"world!"
+bounded_string_value_default4: "Hello'world!"
+bounded_string_value_default5: "Hello\"world!"
+)",
+      to_yaml(
+        msg).c_str());
+  }
+
+  {
+    rosidl_generator_cpp::msg::WStrings msg;
+    msg.wstring_value = u"Hello\nwörld";
+    EXPECT_STREQ(
+      R"(wstring_value: "Hello
+w\xf6rld"
+wstring_value_default1: "Hello world!"
+wstring_value_default2: "Hell\xf6 w\xf6rld!"
+wstring_value_default3: "\u30cf\u30ed\u30fc\u30ef\u30fc\u30eb\u30c9"
+array_of_wstrings:
+- ""
+- ""
+- ""
+bounded_sequence_of_wstrings: []
+unbounded_sequence_of_wstrings: []
+)",
+      to_yaml(
+        msg).c_str());
+  }
+
+  /*{
+    test_msgs::idl::IdlOnlyTypes msg;
+    msg.wchar_value = u'ö';
+    msg.long_double_value = 1.125;
+    EXPECT_STREQ(
+      R"(wchar_value: "\u00f6"
+long_double_value: 1.12500
+)",
+      to_yaml(
+        msg).c_str());
+
+    msg.wchar_value = u'貓';
+    EXPECT_STREQ(
+      R"(wchar_value: "\u8c93"
+long_double_value: 1.12500
+)",
+      to_yaml(
+        msg).c_str());
+  }*/
```

Comment:
Because the `test_interface_files` package doesn't export this file atm: https://github.com/ros2/test_interface_files/blob/c4d710e3394ad09a71a257f5490688653fb441b3/CMakeLists.txt#L32